crates/burn-wgpu/src/template/reduction/recursive_sum.wgsl

```diff
@@ -21,14 +22,23 @@ fn main(
 ) {
     let id_global = global_id.y * (num_workgroups.x * WORKGROUP_SIZE_X) + global_id.x;
     let id_local = local_id.y * WORKGROUP_SIZE_X + local_id.x;
+    let last_index = (WORKGROUP_SIZE_Y * num_workgroups.y) * (WORKGROUP_SIZE_X * num_workgroups.x) - 1u;
 
     data[id_local] = input[id_global];
 
     workgroupBarrier();
 
+    var num_data = WORKGROUP_SIZE;
+    if (id_local == last_index) {
+        num_data = min(WORKGROUP_SIZE, arrayLength(&input) % WORKGROUP_SIZE);
+        if (num_data == 0u) {
+            num_data = WORKGROUP_SIZE;
+        }
+    }
+
```

Comment:
We are going to remove that kernel pretty soon and replace it with the same ones as the reduce_dim (two version).

Reply:
no worries at all. helped me ramp up on the codebase. thanks for looking into it.